great_expectations/core/batch.py

```diff
@@ -261,7 +271,22 @@ def to_json_dict(self) -> dict:
         make this refactoring infeasible at the present time.
         """
         dict_obj: dict = self.to_dict()
+
+        # if a Pandas or Spark DF appears in BatchRequest, temporarily replace it with
+        # str placeholder before calling convert_to_json_serializable so that
+        # DF is not serialized
+        batch_data = dict_obj.get("runtime_parameters", {}).get("batch_data")
```

Comment:
@roblim The idea is good, but we have a better way!  In `batch.py`, there is `batch_request_contains_batch_data()`, which you can use (please use the keyword-style argument to it, thanks!).

Reply:
@roblim In addition, if all you care about is to prevent serializing a dataframe, why not just say:
```
if batch_request_contains_batch_data(batch_request=batch_request):
   batch_data: Union[BatchRequest, RuntimeBatchRequest, dict] = batch_request["runtime_parameters"]["batch_data"] = str(type(batch_request["runtime_parameters"]["batch_data"]))
   serializeable_dict: dict = convert_to_json_serializable(data=get_batch_request_as_dict(batch_request=batch_request))
 batch_request["runtime_parameters"]["batch_data"] = batch_data

return serializeable_dict
```
the point being is that you should not even have to run `self.to_dict()` -- please double-check me on that by getting the code snippet above to work for your purposes.  Thanks!

Reply:
Thanks for the excellent feedback @alexsherstinsky! Comment addressed in [1448454](https://github.com/great-expectations/great_expectations/pull/4352/commits/14484543d78d95a7a195674055c2365e3baec203).